src/dodal/devices/zocalo/zocalo_results.py

```diff
@@ -80,15 +79,17 @@ def __init__(
         self._raw_results_received: Queue = Queue()
         self.transport: CommonTransport | None = None
 
-        self.results = create_soft_signal_r(list[XrcResult], "results", self.name)
-        self.centres_of_mass = create_soft_signal_r(
+        self.results, _ = soft_signal_r_and_setter(
+            list[XrcResult], "results", self.name
+        )
+        self.centres_of_mass, _ = soft_signal_r_and_setter(
             NDArray[np.uint64], "centres_of_mass", self.name
         )
-        self.bbox_sizes = create_soft_signal_r(
+        self.bbox_sizes, _ = soft_signal_r_and_setter(
             NDArray[np.uint64], "bbox_sizes", self.name
         )
-        self.ispyb_dcid = create_soft_signal_r(int, "ispyb_dcid", self.name)
-        self.ispyb_dcgid = create_soft_signal_r(int, "ispyb_dcgid", self.name)
+        self.ispyb_dcid, _ = soft_signal_r_and_setter(int, "ispyb_dcid", self.name)
+        self.ispyb_dcgid, _ = soft_signal_r_and_setter(int, "ispyb_dcgid", self.name)
```

Comment:
Must: Again, the signature is different

Reply:
This really ought to be caught by a test

Reply:
https://github.com/DiamondLightSource/dodal/issues/527